datafusion/sql/src/unparser/dialect.rs

```diff
@@ -45,6 +45,12 @@ pub trait Dialect {
     fn interval_style(&self) -> IntervalStyle {
         IntervalStyle::PostgresVerbose
     }
+
+    // Does the dialect use DOUBLE PRECISION to represent Float64 rather than DOUBLE?
+    // E.g. Postgres uses DOUBLE PRECISION instead of DOUBLE
+    fn use_double_precision_for_float64(&self) -> bool {
```

Comment:
Similarly to https://github.com/apache/datafusion/pull/11494#discussion_r1680046876 what do you think about making this simply return the AST type directly?

```suggestion
    fn use_double_precision_for_float64(&self) -> sqlparser::ast::DataType {
```

That is likely a more flexible API -- this formulation seems very narrow for postgres vs MySQL

Reply:
Hey @alamb , thanks for the feedback! Directly return AST type seems clearer and more flexible to me. I update the changes in my newest commit, please take a look when you are available:
* Return `sqlparser::ast::DataType` instead of `bool`
* Change the method name from `use_double_precision_for_float64` to `float64_ast_dtype`